lib/cisco_node_utils/feature.rb

```diff
@@ -19,6 +19,31 @@
 module Cisco
   # Feature - node util class for managing common features
   class Feature < NodeUtil
+    def self.fabric_enable
+      # install feature-set and enable it
+      return if fabric_enabled?
+      config_set('feature', 'fabric', state: 'install') unless fabric_installed?
+      config_set('feature', 'fabric', state: '')
+    end
+
+    def self.fabric_enabled?
+      config_get('feature', 'fabric') =~ /^enabled/
+    end
+
+    def self.fabric_installed?
+      config_get('feature', 'fabric') !~ /^uninstalled/
+    end
+
+    def self.fabric_forwarding_enable
+      return if fabric_forwarding_enabled?
+      Feature.fabric_enable unless node.product_id =~ /N9/
```

Comment:
Ugh. So N9 supports `feature fabric forwarding` but not `feature-set fabric` ? blech.

I don't like platform checks in the code itself unless it can't be helped. Consider doing:

```
fabric:
  _exclude: [/N(3|9)/]
  config_get: "show feature-set"
  config_get_token: '/^fabric[\s\d]+(\w+)/'
  config_set: "<state> feature-set fabric"
```

```
  def self.fabric_supported?
    config_get('feature', 'fabric')
  end

  def self.fabric_forwarding_enable
    return if fabric_forwarding_enabled?
    Feature.fabric_enable if Feature.fabric_supported?
     config_set('feature', 'fabric_forwarding')
  end
```